src/interfaces/chain.cpp

```diff
@@ -35,6 +150,34 @@ class ChainImpl : public Chain
         return std::move(result);
     }
     std::unique_ptr<Chain::Lock> assumeLocked() override { return MakeUnique<LockImpl>(); }
+    bool findBlock(const uint256& hash, CBlock* block, int64_t* time, int64_t* time_max) override
+    {
+        CBlockIndex* index;
+        {
+            LOCK(cs_main);
+            auto it = ::mapBlockIndex.find(hash);
+            if (it == ::mapBlockIndex.end()) {
+                return false;
+            }
+            index = it->second;
```

Comment:
in commit 22abf4a61a Add findFork and findBlock to the Chain interface


Same here (use `LookupBlockIndex`)